### PR TITLE
feat(alarms): remove request count alarm and make default alarms optional

### PR DESCRIPTION
**Pull request by kkyeboah**
## Goal

- When default alarms `httpLatency` and `http5xxError` properties are not defined, skip creating any alarms.
- Remove `httpRequestCount` alarm, if needed, it can be added as a custom alarm
- Rename `http5xxError` to `http5xxErrorPercentage`

## Reference

Tickets: 
- BACK-1068
- BACK-1054

### CODE REVIEW

File: src/pocket/PocketALBApplication.ts [hunk 1]
```diff
@@ -737,103 +716,86 @@ export class PocketALBApplication extends Resource {
...
+    if (alarmsConfig?.http5xxError) defaultAlarms.push(http5xxAlarm);
+
+    if (alarmsConfig?.httpLatency) defaultAlarms.push(latencyAlarm);
 
     if (alarmsConfig?.customAlarms) {
       defaultAlarms.push(...alarmsConfig.customAlarms);
     }
 
-    this.createAlarms(defaultAlarms);
+    if (defaultAlarms.length) this.createAlarms(defaultAlarms);
```
Comment (jpetto) [inline src/pocket/PocketALBApplication.ts:798]: do we need the `if` here? `createAlarms` does a `forEach`... doesn't hurt i guess.

File: src/pocket/PocketALBApplication.ts [hunk 2]
```diff
@@ -737,103 +716,86 @@ export class PocketALBApplication extends Resource {
     const evaluationPeriods = {
       http5xxError: alarmsConfig?.http5xxError?.evaluationPeriods ?? 5,
       httpLatency: alarmsConfig?.httpLatency?.evaluationPeriods ?? 1,
-      httpRequestCount: alarmsConfig?.httpRequestCount?.evaluationPeriods ?? 1,
+    };
+    const http5xxAlarm: CloudwatchMetricAlarmConfig = {
```
Comment (jpetto) [inline src/pocket/PocketALBApplication.ts:720]: not to scope creep, but there is another breaking change ticket open for this repo. i don't know how much we care about releasing consecutive/multiple breaking changes updates.

the ticket is to rename `http5xxError` to a name that clearly communicates it's a percentage based error (and not a count based error): https://getpocket.atlassian.net/browse/BACK-1054

**Review by jpetto — DISMISSED** [review]
a couple non-blocking comments. code changes look good.

File: src/pocket/PocketALBApplication.ts (hunk 2 above)
Comment (kkyeboah) [reply src/pocket/PocketALBApplication.ts:720]: @jpetto I can add that here

**Review by jpetto — APPROVED** [review]
yay! 🎉
